### PR TITLE
[apex,vf] Fix #1648 - Remove CodeClimate dependency

### LINKED ISSUE #1648: [apex,vf] Remove CodeClimate dependency
Applies to all rules related to Apex and Visualforce. Back then when I ported PDM to Salesforce CodeClimate was the only platform to integrate with. To make PMD work with them we had to add custom renderers and properties.

Neighter I nor somebody else is using this anymore and it clutters the rule and ruleset code. So I'd suggest we remove this again. 

I can provide a PR for that. But I want to hear your opinion on this @jsotuyod @adangel

### PR DESCRIPTION
**Pull request by rsoesemann**

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Running pmdtester failed, this message is mainly used to remind the maintainers of PMD.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: pmd-apex/src/main/resources/rulesets/apex/ruleset.xml [hunk 1]
```diff
@@ -1,444 +1,186 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<ruleset xmlns="http://pmd.sourceforge.net/ruleset/2.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" name="Default ruleset used by the CodeClimate Engine for Salesforce.com Apex" xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 https://pmd.sourceforge.io/ruleset_2_0_0.xsd">
-   <description>Default ruleset used by the Code Climate Engine for Salesforce.com Apex</description>
+<ruleset xmlns="http://pmd.sourceforge.net/ruleset/2.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" name="Default ruleset for Salesforce.com Apex" xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 https://pmd.sourceforge.io/ruleset_2_0_0.xsd">
+   <description>Default ruleset for Salesforce.com Apex</description>
```
Comment (oowekyala) [inline pmd-apex/src/main/resources/rulesets/apex/ruleset.xml]: This ruleset should probably be renamed `quickstart.xml` for consistency with the Java one. Not in this PR though. I'll open an issue to track that.

File: pmd-core/src/main/java/net/sourceforge/pmd/renderers/CodeClimateRule.java [hunk 2]
```diff
@@ -53,45 +47,4 @@ public String toString() {
             return result;
         }
     }
-
```
Comment (oowekyala) [inline pmd-core/src/main/java/net/sourceforge/pmd/renderers/CodeClimateRule.java]: CodeClimateRule doesn't appear to be used anymore, not even by the renderer. So, the interface can probably be removed entirely and the CodeClimateCategory enum be moved into the CodeClimateRenderer?

File: docs/pages/7_0_0_release_notes.md [hunk 3]
```diff
@@ -22,6 +22,8 @@ This is a {{ site.pmd.release_type }} release.
 ### Fixed Issues
 
 ### API Changes
+*   [#1648](https://github.com/pmd/pmd/pull/1702): \[apex,vf] Remove CodeClimate dependency - [Robert Sösemann](https://github.com/rsoesemann)
```
Comment (oowekyala) [inline docs/pages/7_0_0_release_notes.md]: The fact that some properties were removed and their names should be mentioned here, probably with a short mention of the reason. The link to the issue can be mentioned for more details, but we try to explain how users may be affected by API changes in the release notes directly, without having to open an external link to find out what changed.

File: pmd-apex/src/main/resources/rulesets/apex/ruleset.xml (hunk 1 above)
Comment (rsoesemann) [reply pmd-apex/src/main/resources/rulesets/apex/ruleset.xml]: No need for an extra issue. I'll do that in my corrections.

Comment (rsoesemann) [reply pmd-apex/src/main/resources/rulesets/apex/ruleset.xml]: Regarding your above comments:

- This PR brings PMD where it was before CodeClimate wise. I never heard that anyone in the Salesforce world uses it so its fair to clean up the mess I made. I have no interest in improving anything Code Climate related. Let that do others...
- I will add the additional consequences to the release notes doc.

**Comment by rsoesemann** [comment]
Comments to your overall comment:

- This PR just removed stuff from PMD that I fought hard with Andreas in 2016 to get it in. I'm just cleaning my mess. Nobody in the Salesforce world uses Code Climate. So lets get rid of tech debt and waste. I have no interest at all in supporting them more by adding new concepts.

- I will add a text to the release notes.

- Can you help me once with merging my stuff to the master. Maybe on Thursday via Gitter chat. Never did it and I guess there are some pitfalls and tricks.

**Comment by rsoesemann** [comment]
I accidentally "resolved" your comment by renaming the description of the ruleset. I don't want to do it because it is not a quickstart but the default that every starter should use. There are not that many and no controversial rules so its simpler to just say: Bäm thats the default. Use it!

File: pmd-apex/src/test/resources/net/sourceforge/pmd/lang/apex/rule/codestyle/xml/MethodNamingConventions.xml [hunk 4]
```diff
@@ -68,8 +68,8 @@ public class Foo {
     
     <test-code>
         <description>#1573 method names should not contain underscores, but skip test methods 1</description>
-        <expected-problems>0</expected-problems>
         <rule-property name="skipTestMethodUnderscores">true</rule-property>
+        <expected-problems>0</expected-problems>
```
Comment (adangel) [inline pmd-apex/src/test/resources/net/sourceforge/pmd/lang/apex/rule/codestyle/xml/MethodNamingConventions.xml]: I don't understand this change: Why is this necessary, or did we change the rule-test schema?

**Comment by adangel** [comment]
> Can you help me once with merging my stuff to the master.

We won't merge from 7.0.0 -> master, but only from master -> 7.0.0.
So, what we need on master is a separate PR that:
* Deprecates CodeClimateRule (if not already done)
* Deprecates the rule properties (if not already done)
* Update the api changes section in the release notes, the these classes/constants are deprecated for removal.

Ideally, we'll do this change on master first before we merge this PR to 7.0.0.

File: docs/pages/7_0_0_release_notes.md [hunk 5]
```diff
@@ -22,6 +22,9 @@ This is a {{ site.pmd.release_type }} release.
 ### Fixed Issues
 
 ### API Changes
+* [#1648](https://github.com/pmd/pmd/pull/1702): \[apex,vf] Remove CodeClimate dependency - [Robert Sösemann](https://github.com/rsoesemann)
+  Properties "cc_categories", "cc_remediation_points_multiplier", "cc_block_highlighting" can no longer be overridden in rulesets. 
+  They were deprecated without replacement.
```
Comment (adangel) [inline docs/pages/7_0_0_release_notes.md]: We should also mention the change on the CodeClimateRenderer:

> The CodeClimate renderer now only outputs default values for category and remediation points.
> The only mapping inside PMD existed for Apex rules which is now removed.
> The mapping between PMD rule's categories and CodeClimate categories needs to be maintained separately
> in the [CodeClimate engine for PMD](https://github.com/codeclimate/codeclimate-pmd).

**Comment by adangel** [comment]
Btw, it seems like Apex support was only added in November 2018 to code  climate (see https://github.com/codeclimate/codeclimate-pmd/commit/5455b68ad66a3e211b9a39320e69cf4e5cd83aa7)... for Java, I assume, they just use the default category/remediation points...

I see, your plugin https://docs.codeclimate.com/docs/apexmetrics is deprecated in favor of the more general https://docs.codeclimate.com/v1.0/docs/pmd

File: pmd-apex/src/test/resources/net/sourceforge/pmd/lang/apex/rule/codestyle/xml/MethodNamingConventions.xml (hunk 4 above)
Comment (rsoesemann) [reply pmd-apex/src/test/resources/net/sourceforge/pmd/lang/apex/rule/codestyle/xml/MethodNamingConventions.xml]: The builds failed here because the order was wrong. I have no clue why this didn’t fail in the past but I fixed it for this pr

**Comment by rsoesemann** [comment]
Ok as I understand @adangel I will add a second PR for master PR first before this will be merged.

File: pmd-apex/src/test/resources/net/sourceforge/pmd/lang/apex/rule/codestyle/xml/MethodNamingConventions.xml (hunk 4 above)
Comment (adangel) [reply pmd-apex/src/test/resources/net/sourceforge/pmd/lang/apex/rule/codestyle/xml/MethodNamingConventions.xml]: We seem to run into this problem: https://github.com/junit-team/junit5/issues/987

Comment (adangel) [reply pmd-apex/src/test/resources/net/sourceforge/pmd/lang/apex/rule/codestyle/xml/MethodNamingConventions.xml]: #1707

**Comment by jsotuyod** [comment]
I see we deprecated [`CodeClimateRule`](https://github.com/pmd/pmd/blob/master/pmd-core/src/main/java/net/sourceforge/pmd/renderers/CodeClimateRule.java), but we didn't do so with [`CodeClimateRenderer`](https://github.com/pmd/pmd/blob/master/pmd-core/src/main/java/net/sourceforge/pmd/renderers/CodeClimateRenderer.java)… we should do so ASAP.

**Comment by adangel** [comment]
See also: https://github.com/pmd/pmd/issues/1648#issuecomment-467816344

But true, as soon we are going to remove the code climate specific properties, the category and remediation points will fall back to default....
So, the code climate renderer can still be used, even without having a CodeClimateRule.
